libnmstate/nm/route.py

```diff
@@ -141,3 +147,40 @@ def _get_default_route_config(gateway, metric, default_table_id, iface_name):
         Route.NEXT_HOP_ADDRESS: gateway,
         Route.METRIC: metric,
     }
+
+
+def add_routes(setting_ip, routes):
+    default_gateway_defined = False
+    for route in routes:
+        if route[Route.DESTINATION] in (IPV4_DEFAULT_GATEWAY_DESTINATION,
+                                        IPV6_DEFAULT_GATEWAY_DESTINATION):
+            if default_gateway_defined:
+                raise NmstateNotImplementedError(
+                    'Only a single default gateway is supported.')
```

Comment:
Perhaps, to avoid questions, mention that this is a per profile limitation.
I would also mention this in the commit message and in tests to show this.